go/worker/keymanager/watcher.go

```diff
@@ -90,23 +103,21 @@ func (knw *kmNodeWatcher) watchNodes() {
 			peers[n.P2P.ID] = struct{}{}
 		}
 
-		knw.w.accessList.UpdateNodes(knw.w.runtimeID, nodes)
-		if pm := knw.w.commonWorker.P2P.PeerManager(); pm != nil {
```

Comment:
Anyone knows why was this check useful and if it can be removed?

Reply:
My only thought would be maybe when the no-op p2p is in use for archive nodes? But not sure if key manager stuff can even be started in that case? Maybe this is no longer needed since you cannot configure a key manager node in archive mode as the mode can be either archive or key manager.

Reply:
You are right. The no-op p2p is used only for archive nodes as the `init` method disables the key manager in this case. So it is currently safe to remove this check.